kubedb.com/v1alpha2/redissentinel_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright AppsCode Inc. and Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"testing"
+
+	mt "kmodules.xyz/resource-metrics/testing"
+
+	"github.com/google/go-cmp/cmp"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestRedisSentinel(t *testing.T) {
+	type want struct {
+		replicas       int64
+		mode           string
+		totalResources core.ResourceRequirements
+		appResources   core.ResourceRequirements
+	}
+	tests := []struct {
+		name string
+		want want
+	}{
+		{
+			name: "testdata/kubedb.com/v1alpha2/redissentinel/cluster.yaml",
+			want: want{
+				replicas: 3,
+				mode:     DBModeCluster,
+				totalResources: core.ResourceRequirements{
+					Limits: core.ResourceList{
+						core.ResourceCPU:     resource.MustParse("3"),
+						core.ResourceMemory:  resource.MustParse("3456Mi"),
+						core.ResourceStorage: resource.MustParse("3Gi"),
+					},
+					Requests: core.ResourceList{
+						core.ResourceCPU:     resource.MustParse("1.5"),
+						core.ResourceMemory:  resource.MustParse("1692Mi"),
+						core.ResourceStorage: resource.MustParse("3Gi"),
+					},
+				},
+				appResources: core.ResourceRequirements{
+					Limits: core.ResourceList{
+						core.ResourceCPU:     resource.MustParse("1500m"),
+						core.ResourceMemory:  resource.MustParse("3Gi"),
+						core.ResourceStorage: resource.MustParse("3Gi"),
+					},
+					Requests: core.ResourceList{
+						core.ResourceCPU:     resource.MustParse("750m"),
+						core.ResourceMemory:  resource.MustParse("1500Mi"),
+						core.ResourceStorage: resource.MustParse("3Gi"),
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			obj, err := mt.Load(tt.name)
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			c := RedisSentinel{}.ResourceCalculator()
+
+			if got, err := c.Replicas(obj); err != nil {
+				t.Errorf("Replicas() error = %v", err)
+			} else if got != tt.want.replicas {
+				t.Errorf("Replicas found = %v, expected = %v", got, tt.want.replicas)
+			}
+
+			if got, err := c.Mode(obj); err != nil {
```

Comment:
May be need to remove this one also?

Reply:
We still want to test the function is returning correct value.